src/jit/morph.cpp

```diff
@@ -11378,6 +11378,15 @@ GenTree* Compiler::getSIMDStructFromField(GenTree*   tree,
                 *simdSizeOut          = simdNode->gtSIMDSize;
                 *pBaseTypeOut         = simdNode->gtSIMDBaseType;
             }
+#ifdef FEATURE_HW_INTRINSICS
+            else if (obj->OperIsSimdHWIntrinsic())
```

Comment:
Should we merge this with the `GT_SIMD` check above (given that `GenTreeHWIntrinsic` and `GenTreeSIMD` both inherit from `GenTreeJitIntrinsic`, which is the node that actually carries this data)?

Reply:
> inherit from GenTreeJitIntrinsic, which is the node that actually carries this data)?

Yes, but we do not have `AsJitIntrinsic`.

Reply:
Might be worth investigating if its worthwhile to add later (maybe as part of https://github.com/dotnet/coreclr/issues/15641)

Reply:
> if its worthwhile to add later (maybe as part of #15641)

Agree.